components/DateInput.tsx

```diff
@@ -0,0 +1,49 @@
+import * as React from 'react';
+import { Controller, Control } from 'react-hook-form';
+import { LocalizationProvider, DatePicker, DatePickerProps } from '@mui/x-date-pickers';
+import { AdapterDayjs } from '@mui/x-date-pickers/AdapterDayjs';
+import dayjs from 'dayjs';
+
+type DateInputProps = Omit<DatePickerProps<dayjs.Dayjs>, 'value' | 'onChange'> & {
+  name: string;
+  control: Control;
+  rules?: object;
+  defaultValue?: string;
+};
+
+const DateInput = ({
+  name,
+  control,
+  label,
+  rules = {},
+  defaultValue,
+  ...rest
+}: DateInputProps) => (
+  <LocalizationProvider dateAdapter={AdapterDayjs}>
+    <Controller
+      name={name}
+      control={control}
+      rules={rules}
+      defaultValue={defaultValue}
+      render={({ field, fieldState }) => (
+        <DatePicker
+          label={label}
+          value={field.value ?? null}
```

Comment:
Does `null` value solve your issue for error regarding uncontrolled input turning into controlled?

Reply:
YEs, with this fix I dont get any more that error